spec/executor_spec.lua

```diff
@@ -1,42 +1,48 @@
 local executor = require 'apicast.executor'
-local policy_chain = require 'apicast.policy_chain'
+local PolicyChain = require 'apicast.policy_chain'
 local Policy = require 'apicast.policy'
 
 describe('executor', function()
-  local phases = {
-    'init', 'init_worker',
-    'rewrite', 'access', 'balancer',
-    'header_filter', 'body_filter',
-    'post_action',  'log'
-  }
+  local function policy_mt(policy)
+    return getmetatable(policy).policy
+  end
 
   it('forwards all the policy methods to the policy chain', function()
-    -- Policies included by default in the executor
-    local default_executor_chain = {
-      require 'apicast.policy.load_configuration',
-      require 'apicast.policy.find_service',
-      require 'apicast.policy.local_chain'
-    }
-
+    local chain = PolicyChain.default()
+    local exec = executor.new(chain)
     -- Stub all the nginx phases methods for each of the policies
-    for _, phase in ipairs(phases) do
-      for _, policy in ipairs(default_executor_chain) do
+    for _, phase in Policy.phases() do
+      for _, policy in ipairs(chain) do
         stub(policy, phase)
       end
     end
 
     -- For each one of the nginx phases, verify that when called on the
     -- executor, each one of the policies executes the code for that phase.
-    for _, phase in ipairs(phases) do
-      executor[phase](executor)
-      for _, policy in ipairs(default_executor_chain) do
+    for _, phase in Policy.phases() do
```

Comment:
I was going to say that the test would be incomplete after this change, but I see that in the specs of policy we're already testing that `Policy.phases()` returns the correct phases. Nice cleanup :+1:

Reply:
Actually the test was incomplete before this change :) Because It didn't had `content`. I think there might be one more place where it has own list.